### PR TITLE
Feat tests

### PR DESCRIPTION
**Pull request by khlam**
# Description of changes:
Initial tests, small refactor

# Action List:
- [x] I've bumped `package.json` version (minor or major.)
- [x] I've kept this branch up-to-date with `dev`.

### CODE REVIEW

File: src/__tests__/parsePage.spec.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { parseAddonDetails } from '../parsePage'
+
+describe('parseAddonDetails function', () => {
+    describe('WHEN making a REAL request to curseforge.com', () => {
+        describe('WHEN using a real URLObj for DBM hosted on Curseforge', () => { // TODO: Sometimes request fails, need to mock websocket and website
```
[the remark this answers is not in the document]
Comment (khlam) [reply src/__tests__/parsePage.spec.js]: to be adjusted/fixed in later PR

File: src/integrityCheck.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { mainWindow } from '../main'
+
+// Verifies that all addon subdirs are present. Updates addon statuses accordingly
+export function integrityCheck (installedAddonsDict, configObj) {
+    let changed = false
+    console.log('Checking install integrity')
+    fetchAllAddonDirs(configObj.addonDir)
+      .then(dirList => {
+        for (const [name] of Object.entries(installedAddonsDict)) { // Checks if addon is installed by making sure all subdirs are found in dirlist     
+          if (installedAddonsDict[name].subdirs.every(val => dirList.includes(val)) !== true) {
```
[the remark this answers is not in the document]
Comment (khlam) [reply src/integrityCheck.js]: to be refactored in later PR

File: src/__tests__/config.spec.js [hunk 3]
```diff
@@ -38,5 +38,254 @@ describe('initConfig function', () => {
...
+          return promise.then(result => { 
+            expect(result).toEqual({})
+          }) 
+        })
+      })
+      
+      describe('GIVEN an addon record file exists and it is not empty', () => {
+        const mockAddonRecordFile = 
+        {
+          "a-good-wow-addon": {
```
[the remark this answers is not in the document]
Comment (khlam) [reply src/__tests__/config.spec.js]: fixed

File: src/__tests__/config.spec.js [hunk 4]
```diff
@@ -38,5 +38,254 @@ describe('initConfig function', () => {
       })
     })
   })
-  
+})
+
+describe('readAddonList function ', () => {
+  describe('WHEN reading an addon record file (addons.js) ', () => {
+    describe('GIVEN a config exists', () => {
+      const mockConfig = {
```
[the remark this answers is not in the document]
Comment (khlam) [reply src/__tests__/config.spec.js]: fixed

**Comment by khlam** [comment]
FYI 70df98a contains breaking change for all older versions, changes dictionary key "URL" to lowercase ("url") for consistency. WoA may launch with errors, uninstall/Install all addons to fix and avoid conflicts in addons.js.

File: src/__tests__/uninstallAddon.spec.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import { mockAddonObj, mockConfig, mockInstallAddonsDict } from './mock'
+import { uninstallAddon } from '../uninstallAddon'
+import fs from 'fs'
+
+describe('uninstallAddon function', () => {
+    describe('GIVEN a valid configObj', () => {
+        describe('GIVEN a valid installAddonsDict', () => {
```
Comment (zainkai) [inline src/__tests__/uninstallAddon.spec.js]: `And a valid installAddonsDict`

File: src/integrityCheck.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import { saveToAddonList } from './config'
+import { mainWindow } from '../main'
+
+// Verifies that all addon subdirs are present. Updates addon statuses accordingly
+export function integrityCheck (installedAddonsDict, configObj) {
+    let changed = false
+    console.log('Checking install integrity')
+    fetchAllAddonDirs(configObj.addonDir)
+      .then(dirList => {
+        for (const [name] of Object.entries(installedAddonsDict)) { // Checks if addon is installed by making sure all subdirs are found in dirlist
```
Comment (zainkai) [inline src/integrityCheck.js]: `for of` is meant for iterables (arrays), `for in` is for enumerables (objects). But your solution works, its just extra steps

Comment (khlam) [reply src/integrityCheck.js]: integrityCheck.js to be refactored

File: src/__tests__/uninstallAddon.spec.js (hunk 5 above)
Comment (khlam) [reply src/__tests__/uninstallAddon.spec.js]: more tests to be added in when integrityCheck.js is refactored

**Pull request merged**